### PR TITLE
Add Etherscan link to txn sim > details > contract

### PR DESCRIPTION
**Pull request by BrodyHughes**
Fixes: BX-1280

POW:
<video src="https://github.com/rainbow-me/browser-extension/assets/41711440/e97f21ac-121d-4209-b06c-41e2bdd63b32" />

Few more examples with Daniels fixes in:

Screenshot 2024-01-23 at 3 18 38 PM
Screenshot 2024-01-23 at 3 17 43 PM
Screenshot 2024-01-23 at 3 20 48 PM

### CODE REVIEW

File: src/entries/popup/components/AddressDisplay.tsx [hunk 1]
```diff
@@ -47,6 +54,18 @@ function AddressMoreOptions({ address }: { address: Address }) {
             {truncateAddress(address)}
           </Text>
         </DropdownMenuItem>
+        {etherscanLink && (
+          <DropdownMenuItem
+            symbolLeft="doc.text.magnifyingglass"
+            onSelect={() =>
+              goToNewTab({ url: `https://etherscan.io/address/${address}` })
+            }
```
Comment (DanielSinclair) [inline src/entries/popup/components/AddressDisplay.tsx:62]: To support other explorer, we'll want to make this generic. We get explorer URLs upstream from `viem/chains` and you can reference this from the Activity right-click https://github.com/rainbow-me/browser-extension/blob/35e3739098267fab82e67f896719befd156caca1/src/entries/popup/pages/home/Activity/ActivityContextMenu.tsx#L46

File: static/json/languages/en_US.json [hunk 2]
```diff
@@ -1382,6 +1382,7 @@
     "not_available": "Price not available",
     "more_options": {
       "copy_address": "Copy Address",
+      "view_on_etherscan": "View on Etherscan",
```
Comment (DanielSinclair) [inline static/json/languages/en_US.json:1385]: I'd use the key `view_on_explorer` for consistency and accept the explorer name (i.e. Etherscan vs Arbiscan)

**Review by DanielSinclair — CHANGES_REQUESTED** [review]

**Review by DanielSinclair — APPROVED** [review]

**Comment by DanielSinclair** [comment]
Looks good to me!

**Review by brunobar79 — APPROVED** [review]
💯

**Pull request merged**